modules/dhcp_common/free_ips.rb

```diff
@@ -0,0 +1,133 @@
+require 'concurrent'
+require 'logger'
+
+module Proxy::DHCP
+  class FreeIps
+    include ::Proxy::Log
+
+    DEFAULT_ADDRESS_KEEP_TIME = 30*60 # 30 mins
+    DEFAULT_CLEANUP_INTERVAL = 60 # 1 min
```

Comment:
Those two guys must be configurable.

Reply:
Why? Users would need to understand the internals, this is asking for trouble. 
I don't see why anyone would want to change ```DEFAULT_CLEANUP_INTERVAL```. I thought 30 mins is reasonable for ```DEFAULT_ADDRESS_KEEP_TIME```, but can adjust it in either direction.

Reply:
My concern with this would be, that if you have a small subnet a user has no way to see why the IPs are taken and there is no way to "unlock" the IPs without restarting smart-proxy. Feels like trouble to me as well.

Reply:
Again, tinkering with the "CLEANUP_INTERVAL" would require understanding of internal logic, something we cannot demand from users. If 30 mins seems unreasonable, let's arrive at something that works better. We cannot make every parameter we unsure of our users' problem.

Reply:
Maybe it should take the free pool size into account? That way you don't require users to tinker with internal mechanics but can still handle pools with just a few IPs.

Reply:
Well, I don't think we should expose these parameters to users. And I don't want to block this PR. However, I think a UI similar to foreman_dhcp_browser would be in order for Foreman core.

Reply:
@ekohl: pool size size is probably not the metric that should determine the "CLEANUP_INTERVAL" -- it mainly depends on how long it takes a host complete pxe boot + kickstart (or an analogue of it). It's ok if the interval is shorter than the above though, the only side-effect would be an additional delay (due to a failing ping check) when an ip address recommendation is requested.

Reply:
@witlessbird I'd expect that once foreman starts to create the host it'll create a DHCP reservation which means it's no longer available. That'd imply you only need to keep it for as long as the user has the host form open, presses save and foreman creates the reservation.

Reply:
@ekohl: Yes, I'm quite sure this is what happens.

Reply:
Sorry to be late in the discussion.

I am fine not exposing `DEFAULT_CLEANUP_INTERVAL` which you are right is little bit low level, but what is important to have the ability to change is the other one. Depending on many circumstances, your mileage will vary for `DEFAULT_ADDRESS_KEEP_TIME`. If my understanding is correct, this is how long an IP is "put aside" from free pool until user hits Submit button in the UI.

I work with enterprise customers and they sometimes use our project very different way we can't even think of. I can easily imagine someone needs to make a phone call in order to confirm Create host form and 30 minutes might not be enough. Other customer will be booting 250 hosts at one time and that setting is too much for them.

For completeness, I would like to see both as settings with explanation how to set things up. The default values are I think reasonable for regular foreman upstream user, that's fine.